semver.py

```diff
@@ -82,13 +82,40 @@ class VersionInfo:
     :param str prerelease: an optional prerelease string
     :param str build: an optional build string
     """
+    __slots__ = ('_major', '_minor', '_patch', '_prerelease', '_build')
 
     def __init__(self, major, minor, patch, prerelease, build):
-        self.major = major
-        self.minor = minor
-        self.patch = patch
-        self.prerelease = prerelease
-        self.build = build
+        self._major = major
+        self._minor = minor
+        self._patch = patch
+        self._prerelease = prerelease
+        self._build = build
+
+    @property
+    def major(self):
+        return self._major
+
+    @property
+    def minor(self):
+        return self._minor
+
+    @property
+    def patch(self):
+        return self._patch
+
+    @property
+    def prerelease(self):
+        return self._prerelease
+
+    @property
+    def build(self):
+        return self._build
+
+    def __setattr__(self, name, value):
```

Comment:
I think this function is unnecessary

Reply:
I thought also that __setattr__ wasn't necessary...
but not with Python 2.7 and Pypy

Reply:
Why do we want to have private attributes be mutable?

Reply:
You are right ... 
[This build](https://travis-ci.org/k-bx/python-semver/builds/386245547?utm_source=github_status&utm_medium=notification) shows that  `__setattr__` is not required.

Although what is required for Python 2.7 and Pypi is inheriting VersionInfo from `object`.